src/iam.ts

```diff
@@ -272,6 +277,27 @@ export class IAM extends IAMBase {
     throw new Error(ERROR_MESSAGES.DID_DOCUMENT_NOT_INITIALIZED);
   }
 
+  async updateAssetDidDocument(options: {
+    did: string,
+    value: IAttributePayload
+  }) {
+    const { did, value } = options;
+    if (did && this._didSigner) {
+      const defaultValue: IAttributePayload = { publicKey: `0x${new Keys().publicKey}` };
```

Comment:
Key added to document must be controlled by asset owner

Reply:
I don't think this is the case necessarily. Usually will be but not always.

Reply:
So this one can stay as it is, or should i use `this._publicKey` from iam-base class? I'm a little bit confused with this one.

Reply:
No, the key is in the `IUpdateData`, passed in as a param